File: java/src/com/github/yjcpa1_2017/SkeletonJSON.java
package com.github.yjcpa1_2017;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;

import edu.ufl.digitalworlds.j4k.Skeleton;

/**
 *
 * @author aw9223
 */
public class SkeletonJSON {

    @SerializedName("head")
    private Point3D mHead;
    @SerializedName("neck")
    private Point3D mNeck;
    
    @SerializedName("spineShoulder")
    private Point3D mSpineShoulder;
    @SerializedName("spineBase")
    private Point3D mSpineBase;
    @SerializedName("spineMid")
    private Point3D mSpineMid;
    
    @SerializedName("shoulderRight")
    private Point3D mShoulderRight;
    @SerializedName("elbowRight")
    private Point3D mElbowRight;
    @SerializedName("wristRight")
    private Point3D mWristRight;
    @SerializedName("handRight")
    private Point3D mHandRight;
    
    @SerializedName("hipRight")
    private Point3D mHipRight;
    @SerializedName("kneeRight")
    private Point3D mKneeRight;
    @SerializedName("ankleRight")
    private Point3D mAnkleRight;
    @SerializedName("footRight")
    private Point3D mFootRight;
    
    @SerializedName("shoulderLeft")
    private Point3D mShoulderLeft;
    @SerializedName("elbowLeft")
    private Point3D mElbowLeft;
    @SerializedName("wristLeft")
    private Point3D mWristLeft;
    @SerializedName("handLeft")
    private Point3D mHandLeft;
    
    @SerializedName("hipLeft")
    private Point3D mHipLeft;
    @SerializedName("kneeLeft")
    private Point3D mKneeLeft;
    @SerializedName("ankleLeft")
    private Point3D mAnkleLeft;
    @SerializedName("footLeft")
    private Point3D mFootLeft;

    public SkeletonJSON(Skeleton skt) {
        mHead = new Point3D(skt.get3DJoint(Skeleton.HEAD));
        mNeck = new Point3D(skt.get3DJoint(Skeleton.NECK));
        
        mSpineMid = new Point3D(skt.get3DJoint(Skeleton.SPINE_MID));
        mSpineShoulder = new Point3D(skt.get3DJoint(Skeleton.SPINE_SHOULDER));
        mSpineBase = new Point3D(skt.get3DJoint(Skeleton.SPINE_BASE));

        mShoulderRight = new Point3D(skt.get3DJoint(Skeleton.SHOULDER_RIGHT));
        mElbowRight = new Point3D(skt.get3DJoint(Skeleton.ELBOW_RIGHT));
        mWristRight = new Point3D(skt.get3DJoint(Skeleton.WRIST_RIGHT));
        mHandRight = new Point3D(skt.get3DJoint(Skeleton.HAND_RIGHT));

        mHipRight = new Point3D(skt.get3DJoint(Skeleton.HIP_RIGHT));
        mKneeRight = new Point3D(skt.get3DJoint(Skeleton.KNEE_RIGHT));
        mAnkleRight = new Point3D(skt.get3DJoint(Skeleton.ANKLE_RIGHT));
        mFootRight = new Point3D(skt.get3DJoint(Skeleton.FOOT_RIGHT));
        
        mShoulderLeft = new Point3D(skt.get3DJoint(Skeleton.SHOULDER_LEFT));
        mElbowLeft = new Point3D(skt.get3DJoint(Skeleton.ELBOW_LEFT));
        mWristLeft = new Point3D(skt.get3DJoint(Skeleton.WRIST_LEFT));
        mHandLeft = new Point3D(skt.get3DJoint(Skeleton.HAND_LEFT));
        
        mHipLeft = new Point3D(skt.get3DJoint(Skeleton.HIP_LEFT));
        mKneeLeft = new Point3D(skt.get3DJoint(Skeleton.KNEE_LEFT));
        mAnkleLeft = new Point3D(skt.get3DJoint(Skeleton.ANKLE_LEFT));
        mFootLeft = new Point3D(skt.get3DJoint(Skeleton.FOOT_LEFT));
    }
    
    @Override
    public String toString() {
        return new Gson().toJson(this);
    }
}

File: c_sharp/KinectBody/SkeletonJSON.cs

using System;
using System.Collections.Generic;

using Microsoft.Kinect;
using Newtonsoft.Json;
using System.Linq;

namespace KinectBody {

    public class JointJSON {
        public float x;
        public float y;
        public float z;

        public Boolean isTracked;
        public Boolean isTrusted;

        public JointJSON(Joint j) {
            this.x = j.Position.X;
            this.y = j.Position.Y;
            this.z = j.Position.Z;

            this.isTracked = j.TrackingState == TrackingState.Tracked;
            this.isTrusted = j.TrackingState == TrackingState.Inferred;
        }
    }

    public class HandJSON : JointJSON {

        public Boolean isOpened;
        public Boolean isLasso;

        public HandJSON(Joint j, HandState hs) : base(j) {
            isOpened = (hs == HandState.Open);
            isLasso = (hs == HandState.Lasso);
        }
    }

    public class SkeletonJSON {

        public JointJSON head;
        public JointJSON neck;

        public JointJSON spineShoulder;
        public JointJSON spineBase;
        public JointJSON spineMid;

        public JointJSON shoulderRight;
        public JointJSON elbowRight;
        public JointJSON wristRight;
        public JointJSON handRight;

        public JointJSON shoulderLeft;
        public JointJSON elbowLeft;
        public JointJSON wristLeft;
        public JointJSON handLeft;

        public JointJSON hipRight;
        public JointJSON kneeRight;
        public JointJSON ankleRight;
        public JointJSON footRight;

        public JointJSON hipLeft;
        public JointJSON kneeLeft;
        public JointJSON ankleLeft;
        public JointJSON footLeft;

        public SkeletonJSON(Body b) {
            this.head = new JointJSON(b.Joints[JointType.Head]);
            this.neck = new JointJSON(b.Joints[JointType.Neck]);

            this.spineShoulder = new JointJSON(b.Joints[JointType.SpineShoulder]);
            this.spineMid = new JointJSON(b.Joints[JointType.SpineMid]);
            this.spineBase = new JointJSON(b.Joints[JointType.SpineBase]);

            this.shoulderRight = new JointJSON(b.Joints[JointType.ShoulderRight]);
            this.elbowRight = new JointJSON(b.Joints[JointType.ElbowRight]);
            this.wristRight = new JointJSON(b.Joints[JointType.WristRight]);
            this.handRight = new HandJSON(b.Joints[JointType.HandRight], b.HandRightState);

            this.shoulderLeft = new JointJSON(b.Joints[JointType.ShoulderLeft]);
            this.elbowLeft = new JointJSON(b.Joints[JointType.ElbowLeft]);
            this.wristLeft = new JointJSON(b.Joints[JointType.WristLeft]);
            this.handLeft = new HandJSON(b.Joints[JointType.HandLeft], b.HandLeftState);

            this.hipRight = new JointJSON(b.Joints[JointType.HipRight]);
            this.kneeRight = new JointJSON(b.Joints[JointType.KneeRight]);
            this.ankleRight = new JointJSON(b.Joints[JointType.AnkleRight]);
            this.footRight = new JointJSON(b.Joints[JointType.FootRight]);

            this.hipLeft = new JointJSON(b.Joints[JointType.HipLeft]);
            this.kneeLeft = new JointJSON(b.Joints[JointType.KneeLeft]);
            this.ankleLeft = new JointJSON(b.Joints[JointType.AnkleLeft]);
            this.footLeft = new JointJSON(b.Joints[JointType.FootLeft]);
        }

        public override string ToString() {
            return JsonConvert.SerializeObject(this);
        }
    }
}
